lib/net/imap/response_parser.rb

```diff
@@ -1157,6 +1157,7 @@ def body_fields
       def body_fld_param
         return if NIL?
         param = {}
+        shift_token if @token.symbol == T_SPACE
```

Comment:
Do we know if the extra space can also appear before `body-fld-param` when it is `NIL`?

For idiomatic usage, I introduced single character methods to simplify this common pattern (they also allow us to sometimes short-circuit the lexer for a significant performance improvement).  `SP!` is roughly the same as `match(T_SPACE)`, `SP?` is roughly the same as `accept(T_SPACE)`, and we also have `peek_SP?` and `lookahead_SP?`.  See [def_char_matchers](https://github.com/ruby/net-imap/blob/a32cd98673c0292494990aa7662867b13107e036/lib/net/imap/response_parser/parser_utils.rb#L17).

And, most relevant here, I started creating methods or method aliases with a "`quirky_`" prefix, simply to document that we _know_ we're deviating from the spec.  For this, I would use `quirky_SP?`: https://github.com/ruby/net-imap/blob/a32cd98673c0292494990aa7662867b13107e036/lib/net/imap/response_parser.rb#L996-L1000

So, with all of that in mind, what do you think about the following?
```suggestion
        quirky_SP? # See comments on test_bodystructure_extra_space
        return if NIL?
        param = {}
```

Reply:
Thanks for clarifying the idiomatic way of doing things in net-imap! 
Unfortunately I'm not aware if we had such cases when extra space was there when body-fld-param was nil, we don't have such cases documented anywhere. 
I'm gonna commit your suggestion.